packages/core/src/typeschema/validation.ts

```diff
@@ -589,7 +594,19 @@ export function matchDiscriminant(
       if (!value || !sliceElement) {
         return false;
       }
-      if (matchesSpecifiedValue(value, sliceElement)) {
+      if (sliceElement.pattern /*&& deepIncludes(value, sliceElement.pattern)*/) {
+        // return true;
+        return deepIncludes(value, sliceElement.pattern);
+      }
+      if (sliceElement.fixed /*&& deepEquals(value, sliceElement.fixed)*/) {
+        // return true;
+        return deepEquals(value, sliceElement.fixed);
+      }
+
+      if (sliceElement.binding?.strength === 'required' && sliceElement.binding?.valueSet) {
```

Comment:
@mattwiller Based on our offline conversation, I believe this if statement should be changed to:

```typescript
if (sliceElement.binding?.valueSet)
```

Reply:
I think the `required` strength is also necessary here: we don't want to validate optional bindings since technically any other value would also be acceptable